io/EptReader.cpp

```diff
@@ -793,6 +797,45 @@ PointId EptReader::readBinary(PointView& dst, const Key& key,
     return startId;
 }
 
+uint64_t EptReader::readZstandard(PointView& dst, const Key& key,
+        const uint64_t nodeId) const
+{
+#ifdef PDAL_HAVE_ZSTANDARD
+    auto compressed(m_ep->getBinary("ept-data/" + key.toString() + ".zst"));
+    std::vector<char> uncompressed;
+    pdal::ZstdDecompressor dec([&uncompressed](char* pos, std::size_t size)
+    {
+        uncompressed.insert(uncompressed.end(), pos, pos + size);
+    });
+
+    dec.decompress(compressed.data(), compressed.size());
+
+    ShallowPointTable table(
+        *m_remoteLayout,
+        uncompressed.data(),
+        uncompressed.size());
+
+    PointRef pr(table);
+
+    std::lock_guard<std::mutex> lock(m_mutex);
```

Comment:
This seems like a confusing and unnecessarily large mutex scope.  I would think it could go around the `setField()` calls in process().

Reply:
I'm not sure about this - we are appending points to a single `PointView` during each call to `process` and I'd be hesitant to introduce parallelism there.  I personally find this scope much easier to reason about - the multi-threading is used only to acquire chunks (and potentially decompress them into something we can `process`) from potentially remote data sources, and once a chunk is ready for processing we lock and then mutate the output `PointView` in a single-threaded manner.

I made a change that is loosely related to this comment [here](a415808813fd507c95534496ac7a652de429f563), although I did not change the mutex scope.  I think I'd prefer a separate issue be made if you'd like to change how we are threading things here, since the current PR maintains existing behavior of the binary reading logic.